packages/firebase_auth/firebase_auth_web/lib/firebase_auth_web.dart

```diff
@@ -232,9 +226,20 @@ class FirebaseAuthWeb extends FirebaseAuthPlatform {
       String androidPackageName,
       bool androidInstallIfNotAvailable,
       String androidMinimumVersion}) {
-    // TODO(hterkelsen): File issue with `package:firebase` to show
-    // `sendSignInLinkToEmail`.
-    throw UnimplementedError('sendLinkToEmail');
+    final firebase.Auth auth = _getAuth(app);
+    final actionCodeSettings = firebase.ActionCodeSettings(
+      url: url,
+      handleCodeInApp: handleCodeInApp,
+      iOS: firebase.IosSettings(
+        bundleId: iOSBundleID,
+      ),
+      android: firebase.AndroidSettings(
+        packageName: androidPackageName,
+        installApp: androidInstallIfNotAvailable,
+        minimumVersion: androidMinimumVersion,
+      ),
```

Comment:
I was reading the [documentation](https://firebase.google.com/docs/auth/web/passing-state-in-email-actions#passing_statecontinue_url_in_email_actions) and... do we want to kick the user to the Android/iOS app from the web? This change kind of adds a requirement of setting up an iOSBundleID and an androidPackageName on a potentially web-only app.

Can these attributes be passed to the ActionCodeSettings conditionally? Does this handle `null` values gracefully, or does it cause an error?

Reply:
You're right, we shouldn't force users to set up those attributes.

I can change to make this optional but it would require modifying the plugin [to not assert that these attributes are non-null](https://github.com/FirebaseExtended/flutterfire/blob/36b3880179a0334e4d6b347ca6cd2dd0c2c1c600/packages/firebase_auth/firebase_auth/lib/src/firebase_auth.dart#L117) and [update the native code to gracefully handle non-null values](https://github.com/FirebaseExtended/flutterfire/blob/36b3880179a0334e4d6b347ca6cd2dd0c2c1c600/packages/firebase_auth/firebase_auth/android/src/main/java/io/flutter/plugins/firebaseauth/FirebaseAuthPlugin.java#L344) too to keep the api behavior consistent on all platforms.

Is that the route we want to go down or is there an alternative?

Reply:
I'm deferring this one to @hterkelsen, he wrote the original implementation. IMO all these should be optional. I'm guessing in a web-only flow, those are never expected to be used, right?

Reply:
Since these are already required parameters in `package:firebase_auth` I think it makes sense to use them like this. All this is saying is if the Android or iOS app is installed then the sign in link will attempt to open the app,

Reply:
Maybe this is out of the scope of this PR but there is currently no way for users to not include an Android `packageName` or iOS `bundleId`.

Passing a null will cause the method to assert and passing an empty string will cause Firebase to throw an argument error. A possible hack might be to pass in a random string but that seems fragile and possibly dangerous.

I think it would be a good idea to allow users to opt out of certain platforms by passing null.